データベース設計の実施

impress85 opened the pull request:
master branchでpushしてしまい、プルリクエストが出せませんでした。
その為、改めてブランチを作成し、README のタイトルを変更しました。

[ER図](https://gyazo.com/d7f3402812eaa19ac78730d9a88acc8fl)

（参考）
[マスターブランチでプッシュした時の履歴①](https://gyazo.com/b6adcb5532d76197dfb1a7710d62e318)
[マスターブランチでプッシュした時の履歴②](https://gyazo.com/c622d9b78627719329c1123801debb2f)
[マスターブランチでプッシュした時の履歴③](https://gyazo.com/bab55fe64c01bef0ad6531a4d58f1acc)

すみませんが、確認お願いします。

Commit:
データベース設計の実施

master branchでpushしてしまい、プルリクエストが出せませんでした。その為、改めてブランチを作成し、README のタイトルを変更しました。

m85hayashi commented:
ER図が見れなかった為、再添付いたします。
[ER図（再添付）](https://gyazo.com/d7f3402812eaa19ac78730d9a88acc8f)

Katsuno0814 commented on README.md:
```diff
@@ -1,4 +1,4 @@
+# furimaのテーブル設計
 
 ## users テーブル
 
```
| nickname             | string    | null: false                    |
| first_name           | string    | null: false                    |
| first_name_katakana  | char      | null: false                    |
| family_name          | string    | null: false                    |
| family_name_katakana | char      | null: false                    |
| birthday             | date      | null: false                    |
こちら users テーブルにひとまとめにいたしましょう！

Katsuno0814 replied:
## credit_cards テーブル
| Column          | Type      | Options                        |
| --------------- | --------- | ------------------------------ |
| credit_number   | int       | null: false                    |
| expiration_date | int       | null: false                    |
| security_code   | int       | null: false                    |
| transaction_id  | reference | null: false, foreign_key: true |
### Association
- belongs_to :transactions
今回はカードテーブルを用意する必要がないため、削除いたしましょう！

Katsuno0814 replied:
37行目の| image    | ActiveStorage | null: false                    |は
今回の実装では商品出品の画像投稿において「active_storage」を使用するため、設計の段階から削除しておきましょう。
（理由はactive_storage導入時に自動でテーブルなどが生成されるからです）

Katsuno0814 replied:
item_details テーブル
の中身はitemテーブルにまとめましょう！

Katsuno0814 replied:
## transactions テーブル
| Column  | Type      | Options                        |
| ------- | --------- | ------------------------------ |
| user_id | reference | null: false, foreign_key: true |
| item_id | reference | null: false, foreign_key: true |

こちらreferences 型で記述する際は、_idは不要となります。
もしカラム名で_idを使用する予定であれば、integer型を使用した方が良いでしょう。

impress85 replied:
内容修正いたしました。
お手数ですがご確認お願いします。
[修正版ER図](https://github.com/m85hayashi/furima-30550/pull/1)

Katsuno0814 reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

Commit:
READMEとER図の修正

・user_profilesテーブルをusersテーブルへまとめた。
・credit_cardsテーブルの削除
・ActiveStorageを用いるimageの削除
・reference型の_idを削除

m85hayashi commented:
たびたびすみません。
修正版のER図urlに誤りがありました。
再送いたします。

[ER図再修正](https://gyazo.com/35a14b7424769a325048e0d4fd034fd2)

norito0627 commented on README.md:
```diff
@@ -1,91 +1,55 @@
+| family_name          | string | null: false |
+| family_name_katakana | char   | null: false |
+| birthday             | date   | null: false |
 
```
charにしていただいている箇所について、他のカラムと同様にstringで良いかと思います！

impress85 replied:
修正いたしました。

norito0627 commented on README.md:
```diff
@@ -1,91 +1,55 @@
 | shipping_days       | ActiveHash | null: false                    |
-| item_id             | reference  | null: false, foreign_key: true |
+| user                | reference  | null: false, foreign_key: true |
 
```
アクティブハッシュのカラムは_idをつけておきましょう！
また、integer型にしておいていただければ大丈夫です！

impress85 replied:
修正いたしました

norito0627 commented on README.md:
```diff
@@ -96,7 +60,7 @@
 | phone_number   | int        | null: false                    |
-| transaction_id | reference  | null: false, foreign_key: true |
+| transaction    | reference  | null: false, foreign_key: true |
 
```
```
| postal_code    | int        | null: false                    |
| phone_number   | int        | null: false                    |
```
stringにしておきましょう！
integerだと、0が先頭にくると、消えてしまいます！

impress85 replied:
修正いたしました。

norito0627 commented on README.md:
```diff
@@ -96,7 +60,7 @@
 | address        | char       | null: false                    |
 | building_name  | string     | null: true                     |
```
| building_name  | string     | null: true                     |
null: trueは不要です！

impress85 replied:
修正済みです。

norito0627 reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

Commit:
READMEの再修正

m85hayashi commented:
修正いたしました。
[ER図再修正](https://gyazo.com/6d87ac9eab2369f81608cb18169fa0c4)

i-date commented on README.md:
```diff
@@ -1,102 +1,66 @@
-| user_id  | reference     | null: false, foreign_key: true |
+| Column                 | Type       | Options                        |
+| ---------------------- | ---------- | ------------------------------ |
+| item                   | string     | null: false                    |
```
商品名ということが分かりやすくなるように、25行目のカラム名はnameしてはいかがでしょうか？

i-date commented on README.md:
```diff
@@ -1,102 +1,66 @@
+| price                  | int        | null: false                    |
+| describe               | text       | null: false                    |
+| category_id            | integer    | null: false                    |
+| item_condition         | text       | null: false                    |
```
カラム名はテーブル名のitems（というエンティティ）の属性を表すので、カラム名にエンティティを表すitemを付ける必要はありません。
つまり、テーブルの名前（items）から何のconditionか分かるため、カラム名から`item_`を外しましょう。

また、商品状態は基本的に変更されない静的データですので、active_hashを使用します。
そのため、カラム名（`_id`をつける）や型（integer型）を修正してみましょう。
`| condition_id         | integer       | null: false                    |`

impress85 replied:
修正いたしました。

i-date commented on README.md:
```diff
@@ -1,102 +1,66 @@
+| describe               | text       | null: false                    |
+| category_id            | integer    | null: false                    |
+| item_condition         | text       | null: false                    |
+| shipping_cost_payer-id | integer    | null: false                    |
```
30行目の`shipping_cost_payer-id`に`-`が使用されていますので、スネークケースに修正して`shipping_cost_payer_id`としましょう。

impress85 replied:
修正いたしました。

i-date commented on README.md:
```diff
@@ -1,102 +1,66 @@
+| shipping_cost_payer-id | integer    | null: false                    |
+| shipment_source_id     | integer    | null: false                    |
+| shipping_days_id       | integer    | null: false                    |
+| user                   | reference  | null: false, foreign_key: true |
```
33行目の`reference`に`s`を付与して`references`としましょう。

impress85 replied:
修正いたしました。

i-date commented on README.md:
```diff
@@ -1,102 +1,66 @@
-- belongs_to :item
 
 
 ## transactions テーブル
```
こちらは商品の購入を扱うテーブルに設計かと思われますが、ActiveRecordには、あらかじめ「transactionメソッド」が用意されています。
混在や予期せぬエラー発生を避けるため、purchasesやordersといったテーブル名などに変更しておくとよいでしょう。

その際、他テーブルのアソシエーションやカラム名に使用されている`transaction`もあわせて修正いただければと思います。

impress85 replied:
ordersに変更いたしました。

i-date commented on README.md:
```diff
@@ -1,102 +1,66 @@
 | ------- | --------- | ------------------------------ |
-| user_id | reference | null: false, foreign_key: true |
-| item_id | reference | null: false, foreign_key: true |
+| user    | reference | null: false, foreign_key: true |
+| item    | reference | null: false, foreign_key: true |
```
44-45行目の`reference`に`s`を付与して`references`としましょう。

impress85 replied:
修正いたしました。

i-date reviewed (commented):
修正ありがとうございます！
追加コメントいたしましたので、ご確認をお願いいたします。

Commit:
READMEの再再修正

[ER図](https://gyazo.com/8c0a272df04cf2a72d8c37d66f400efd)

impress85 reviewed (commented):
レビューありがとうございます。
全て内容修正させていただきました。
お手数ですがご確認をお願いします。

s-aka-moto commented on README.md:
```diff
@@ -1,102 +1,66 @@
+| Column                 | Type       | Options                        |
+| ---------------------- | ---------- | ------------------------------ |
+| name                   | string     | null: false                    |
+| price                  | int        | null: false                    |
```
こちらも統一のためinteger型としましょう！
`| price                  | int        | null: false                    |`

impress85 replied:
ご指摘ありがとうございます。
表記を統一しました。

s-aka-moto commented on README.md:
```diff
@@ -1,102 +1,66 @@
-| prefectures    | ActiveHash | null: false                    |
+| postal_code    | string     | null: false                    |
+| prefectures_id | integer    | null: false                    |
 | municipality   | char       | null: false                    |
 | address        | char       | null: false                    |
```
これらもstring型にして他のカラムと統一しましょう！
```
| municipality   | char       | null: false                    |
| address        | char       | null: false                    |
```

s-aka-moto replied:
こちらも修正をお願いいたします！

s-aka-moto reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

Commit:
README修正(3)

type表記の統一。

impress85 reviewed (commented):
typeの表記を統一いたしました。

s-aka-moto commented on README.md:
```diff
@@ -1,102 +1,66 @@
-
-### Association
-- belongs_to :user
+- has_many :order
```
has_manyは複数形にしましょう！
`has_many :order`

impress85 replied:
複数形のordersへと修正しました。

s-aka-moto commented on README.md:
```diff
@@ -1,102 +1,66 @@
+| category_id            | integer    | null: false                    |
+| condition_id           | integer    | null: false                    |
+| shipping_cost_payer_id | integer    | null: false                    |
+| shipment_source_id     | integer    | null: false                    |
```
商品購入時にも「都道府県」の情報を使用するため、 `prefectures_id `とすると良いですね！

impress85 replied:
ご指摘ありがとうございます。
prefectures_idへと修正いたしました。

s-aka-moto reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

Commit:
READMEの修正(4)

impress85 reviewed (commented):
内容修正完了しました。
お手数ですが確認お願いします。

s-aka-moto reviewed (commented):
レビュー依頼ありがとうございます！

```
| municipality   | char       | null: false                    |
| address        | char       | null: false                    |

```
こちらもstring型で実装しましょう！

Commit:
READMEの修正(5)

m85hayashi commented:
承知しました。修正いたしました！

s-aka-moto commented:
レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！
引き続き実装頑張っていきましょう！

m85hayashi commented:
どうもありがとうございました！
